docker/package/package_generator.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: 2020 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+import os, sys, subprocess, json
+from distutils.dir_util import copy_tree
+
+class Package:
+    def __init__(self, name, desc):
+        self.name = name
+        self.desc = desc
+
+date = subprocess.check_output(["date", "-R"]).decode().strip()
+meta = json.load(open("../meta.json", "r"))
+tezos_sources = json.load(open("../nix/nix/sources.json", "r"))["tezos"]
+
+common_deps = ["libev-dev", "libgmp-dev", "libhidapi-dev", "m4", "perl", "pkg-config",
+               "wget", "unzip", "rsync"]
+
+active_protocols = json.load(open("../nix/protocols.json", "r"))["active"]
```

Comment:
I kinda dislike that docker way of doing things uses something from `../nix`. I think `protocols.json` can (should) be put outside of `nix/` since it's used not only by nix. There is also `../nix/nix/sources.json`, but it's more complicated, probably it's fine to use as is.

Reply:
Fair enough